src/libstd/io/mod.rs

```diff
@@ -458,7 +458,8 @@ pub trait Read {
     /// All bytes read from this source will be appended to the specified buffer
     /// `buf`. This function will continuously call `read` to append more data to
     /// `buf` until `read` returns either `Ok(0)` or an error of
-    /// non-`ErrorKind::Interrupted` kind.
+    /// non-`ErrorKind::Interrupted` kind. If the size of the input is known in
+    /// advance, performance will be improved by pre-reserving capacity in `buf`.
```

Comment:
Is there a difference between *reserving* and *pre-reserving*? All are things that happen in advance isn't it?